src/app/pages/SearchFunction.ts

```diff
@@ -1,43 +1,53 @@
 import { Result } from '../../__types__';
 
 const MAX_RESULT = 10; // stop searching once we get 20 results
-const MAX_TEXT_LENGTH = 30; // get this many words as a preview
+const MAX_TEXT_LENGTH = 300; // get this many characters as a preview
 
 // return
 // * a string at most <MAX_TEXT_LENGTH> chars long, centered around the keyword position
 // * a boolean value, indicating if this is a placeholder text because the keyword
 //   was not found in the raw text due to some indexing issue
 // A placeholder text maybe replaced later by some other non-placeholder text
 function getShortText(keyword: string, url: string, siteMapDatabase: any): [string, boolean] {
-    const fullText: string = siteMapDatabase[url].text;
+    let fullText: string = siteMapDatabase[url].text;
     if (!fullText) {
         return ['', true];
     }
-    const fullTextArray: string[] = fullText
+
+    fullText = fullText
         .replace(/import .*? from .*?;/gim, '')
         .replace(/\r\n/g, '\n')
         .replace(/\n/gim, ' ')
         .replace(/<!--.*?-->/g, ' ') // replace all comments (including keywords)
         .replace(/\[(.*?)\]\(.*?\)/g, '$1') // replace all the markdown links [text](url) into text
+        .replace(/<FAQExpander question={`(.*?)`}(.*?)>/gim, '$1') // pull out the FAQ question strings <FAQExpander question={''}>
```

Comment:
I can see how this might be extended to other uses, like searching for image captions, or searching for team member bios, or searching for design pattern developer info (if in the future we redesigned that sections to wrap in some components). Is it possible to do some kind of "comment tags" so that this can be applicable to other use cases?

```html
<!-- search: enabled --> blah <!-- search: disabled -->
```

Reply:
I agree that there are probably some more instances where we might like to include stuff in the search index. Not sure how we could effectively do it in a generic way though. The things that are currently getting stripped out are just props inside of HTML tags (we rip out the entire tag). I suppose we could do something like 'stringify all of the props, but I don't know if we really want all of them, probably just a few.

I'm not sure if something like:
```
<MyComponent /* search: enabled */ myProp={'some value'} /* search: disabled */ />
```
would even work. And if it did, would it be more convenient than just adding another condition to the indexer script for the specific prop we want?

Reply:
```
<MyComponent /> some value
```